internal/common/autoscaler/autoscaler.go

```diff
@@ -24,10 +24,6 @@ package autoscaler
 type (
 	AutoScaler interface {
 		Estimator
-		// Acquire X ResourceUnit of resource
-		Acquire(ResourceUnit) error
-		// Release X ResourceUnit of resource
-		Release(ResourceUnit)
```

Comment:
these methods are unnecessary after `Permit` introduction

Reply:
with this removed, this is basically a "periodically update a cache and broadcast" thing, rather than a resource-controller.

which seems fine, just checking that that's what is needed.  without anything using the hooks or `GetCurrent()` I'm kinda struggling to figure out what's needed eventually vs something we should get rid of while we can.